### PR TITLE
Create CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by jfversluis**
### Description of Change ###

Add CONTRIBUTING.md file

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+* Assign values (0,1,2,3) for all enums
+
+### Property Names
+* Include units only if one of the platforms includes it in their implementation. For instance HeadingMagneticNorth implies degrees on all platforms, but PressureInHectopascals is needed since platforms don't provide a consistent API for this.
+
+### Units
+* Use the standard units and most well accepted units when possible. For instance Hectopascals are used on UWP/Android and iOS uses Kilopascals so we have chosen Hectopascals.
+
+### Style
+* Prefer using `==` when checking for null instead of `is`
```
Comment (Depechie) [inline CONTRIBUTING.md:59]: What is the rational behind this? Remember the Equals operator can be overridden... so in essence using .ReferenceEquals(null) is the actual correct check.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+ * APIs should be simple, direct, and generally implemented with static classes and methods whenever practical
+ * New features should have native APIs available to allow implementation on a reasonable subset of the supported platforms, especially  (iOS, Android, UWP)
+ * No new external dependencies should be added to support implementation of new feature work (there can be exceptions but they must be thoroughly considered for the value being added)
+
+#### Proposal States
+##### Open
+Open proposals are still under discussion. Please leave your concrete, constructive feedback on this proposal. +1s and other clutter posts which do not add to the discussion will be removed.
+
+##### Accepted
+Accepted proposals are proposals that both the community and core Xamarin.Essentials agree should be a part of Xamarin.Essentials. These proposals are ready for implementation, but do not yet have a developer actively working on them. These proposals are available for anyone to work on, both community and the core Xamarin.Essentials team.
```
Comment (sthewissen) [inline CONTRIBUTING.md:150]: Essentials a few times.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+If you wish to start working on an accepted proposal, please reply to the thread so we can mark you as the implementor and change the title to In Progress. This helps to avoid multiple people working on the same thing. If you decide to work on this proposal publicly, feel free to post a link to the branch as well for folks to follow along.
+
+###### What "Accepted" does mean
+* Any community member is welcome to work on the idea.
+* The core Xamarin.Essentials team _may_ consider working on this idea on their own, but has not done so until it is marked "In Progress" with a team member assigned as the implementor.
```
Comment (sthewissen) [inline CONTRIBUTING.md:156]: Essentials. And some more in the headings below.

File: CONTRIBUTING.md (hunk 1 above)
Comment (jfversluis) [reply CONTRIBUTING.md:82]: All copy/paste from Essentials

**Pull request merged**